2MASS 0415 Ingest

canavarrete01 opened the pull request:
Short description: Ingest of 2MASS 0415-09. Updates the Burg03 IRTF data within the JSON file, adds the AKARI (Sora12), FIRE (Simc13), KECK I LRIS (Burg04), and KECK I LRIS (Kirk08) data using a new script. 
 

Link to relevant issue: Closes #559 & #375 

For data ingests:
[x] includes script used for ingest
[x ] includes modified JSON files
[ ] Add new tests
[ ] Update the Versions table

kelle commented on data/reference/Instruments.json:
```diff
@@ -453,5 +453,26 @@
         "reference": null
+    },
+    {
+        "instrument": "IRC",
+        "mode": "Missing",
+        "telescope": "AKARI",
+        "description": null,
+        "reference": null
+    },
+    {
+        "instrument": "IRC",
+        "mode": "NIR",
+        "telescope": "AKARI",
+        "description": null,
+        "reference": null
+    },
```
AKARI IRC only needs to be added once.
And use this as a reference: Murakami, H., et al., 2007, The Infrared Astronomical Mission AKARI, PASJ, 59, S369 [[ADS]](http://adsabs.harvard.edu/abs/2007PASJ...59S.369M) [[DOI:10.1093/pasj/59.sp2.S369]](http://dx.doi.org/10.1093/pasj/59.sp2.S369)

canavarrete01 replied:
In what format would suggest for this? Similar to the format for publications (Mura2007) or should I include it as the bibcode?

kelle reviewed (changes requested):
This is really good code but *too* automated for this particular use case. For this situation, do each ingestion one-by-one.

canavarrete01 commented:
Submitted new edits to the ingest script. Updated JSON files as well to include necessary references for AKARI.

canavarrete01 commented:
All test pass now! Updated the following lines in test_data.py to update the number of spectra to be found in the database. 

```
def test_spectra(db):
    regime = "nir"
    t = db.query(db.Spectra).filter(db.Spectra.c.regime == regime).astropy()
    **assert len(t) == 582, f"found {len(t)} spectra in the {regime} regime"**
```

kelle commented on scripts/ingests/Ingest_2MASS_J0415_09.py:
```diff
@@ -0,0 +1,119 @@
+        "regime": "nir",
+        "telescope": "AKARI",
+        "instrument": "IRC",
+        "mode": "NIR",
```
the mode should be IRC04, not NIR

kelle commented on data/reference/Telescopes.json:
```diff
@@ -188,5 +188,10 @@
+    {
+        "telescope": "AKARI",
+        "description": null,
+        "reference": null
```
Apologies, use Mura07 for as the reference for the AKARI telescope.

kelle commented on data/reference/Instruments.json:
```diff
@@ -453,5 +453,12 @@
+        "mode": "NIR",
+        "telescope": "AKARI",
+        "description": null,
+        "reference": "Mura07"
```
Use this publication as the reference for the IRC camera: [10.1093/pasj/59.sp2.S401](https://ui.adsabs.harvard.edu/link_gateway/2007PASJ...59S.401O/doi:10.1093/pasj/59.sp2.S401).

kelle commented on data/reference/Instruments.json:
```diff
@@ -453,5 +453,12 @@
+    },
+    {
+        "instrument": "IRC",
+        "mode": "NIR",
```
Mode is IRC04.

kelle commented on scripts/ingests/Ingest_2MASS_J0415_09.py:
```diff
@@ -0,0 +1,91 @@
+    db,
+    source="2MASS J04151954-0935066",
+    spectrum="https://bdnyc.s3.us-east-1.amazonaws.com/LRIS/KECK_LRIS_2MASSJ04151954-0935066_T8_LRIS_Burg03.fits",
+    regime="nir",
```
should be optical, not nir.

kelle commented on scripts/ingests/Ingest_2MASS_J0415_09.py:
```diff
@@ -0,0 +1,91 @@
+    db,
+    source="2MASS J04151954-0935066",
+    spectrum="https://bdnyc.s3.us-east-1.amazonaws.com/LRIS/KECK_LRIS_2MASS_J04151954-0935066_2006-12-26.fits",
+    regime="nir",
```
should be optical.

kelle commented on tests/test_data.py:
```diff
@@ -246,7 +246,7 @@ def test_spectra(db):
     regime = "nir"
     t = db.query(db.Spectra).filter(db.Spectra.c.regime == regime).astropy()
-    assert len(t) == 578, f"found {len(t)} spectra in the {regime} regime"
+    assert len(t) == 582, f"found {len(t)} spectra in the {regime} regime"
```
Great job getting tests to pass!

kelle reviewed (changes requested):
Code looks great!  Just a few data fixes.

canavarrete01 commented:
Ingested the publication for the IRC instrument. Added details to the AKARI telescope entry. Corrected the regime values for the LRIS KECK spectra to optical from nir.

kelle reviewed (approved):
Looks great!  I'll update the Versions table and make a new release.
